community/git.py

```diff
@@ -0,0 +1,61 @@
+import os
+import os.path
+
+from git.config import GitConfigParser
+import giturlparse
+
+REPO_DIR = os.path.join(
+    os.path.dirname(__file__),
+    '..',
+)
+GIT_CONFIG = os.path.join(
+    REPO_DIR,
+    '.git',
+    'config',
+)
+
+_config = None
+
+
+def get_config():
+    global _config
+    if not _config:
+        _config = GitConfigParser(GIT_CONFIG)
+    return _config
+
+
+def get_config_remote(name='origin'):
+    config = get_config()
+
+    print(config.sections())
+
+    has_remote = False
+
+    for key in config.sections():
+        if key == 'remote "%s"' % name:
+            return config.items(key)
+        elif key.startswith('remote'):
+            has_remote = True
+
+    if has_remote:
+        raise KeyError('No git remote called "%s"' % name)
+
+    raise KeyError('No git remotes found')
+
+
+def get_remote_url():
+    url = os.environ.get('REPOSITORY_URL')
+    if not url:
+        remote = get_config_remote()
+        url = remote[0][1]
+
+    try:
+        url = giturlparse.parse(url)
+    except Exception:
+        url = giturlparse.parse(url + '.git')
```

Comment:
Does this need to be `try/except`ed as well?

Reply:
No, this is a fallback to workaround https://github.com/IonicaBizau/git-url-parse/issues/35

Reply:
No, the try/except was to workaround https://github.com/retr0h/git-url-parse/issues/2